src/app.rs

```diff
@@ -307,10 +308,120 @@ impl App {
             }
             KeyCode::Esc => self.reset_message_selection(),
             KeyCode::Char(c) => self.put_char(c),
+            KeyCode::Tab => {
+                if let Some(idx) = self.data.channels.state.selected() {
+                    self.add_reaction(idx, self.data.input.is_empty()).await;
+                }
+            }
             _ => {}
         }
     }
 
+    pub async fn add_reaction(&mut self, channel_idx: usize, remove: bool) {
+        let emoji_replacer = Replacer::new();
+
+        let input: String = self.data.input.drain(..).collect();
+        let reaction: String = emoji_replacer.replace_all(&input).to_string();
+        let message_reaction = reaction.clone();
+
+        self.data.input_cursor = 0;
+        self.data.input_cursor_chars = 0;
+
+        if let Some(selected_message) = self.data.channels.items[channel_idx].selected_message() {
+            let timestamp = util::utc_now_timestamp_msec();
+            let target_author_uuid = selected_message.from_id;
+            let target_sent_timestamp = selected_message.arrived_at;
+            let mut data_message = DataMessage {
+                body: None,
+                quote: None,
+                reaction: Some(Reaction {
+                    emoji: if !remove {
+                        Some((&reaction).clone())
+                    } else {
+                        Some(
+                            selected_message
+                                .reactions
+                                .iter()
+                                .find(|(uuid, _reac)| *uuid == self.self_id())
+                                .unwrap_or(&(Uuid::from_u128(0), String::from("")))
+                                .1
+                                .clone(),
```

Comment:
If you change the return of this method to `Option<()>` then we can short-circuit it, when there is nothing to remove. Note the `?` in the suggestion.

```suggestion
                            Some(
                            selected_message
                                .reactions
                                .iter()
                                .find(|(uuid, _reac)| *uuid == self.self_id())
                                .map(|(_uuid, reac)| reac.clone())?,
                        )
```

Reply:
Done!